### ISSUE TITLE
Ability to specify a default value for custom option types

### ISSUE THREAD

**Issue by marshall007** [trentm/node-dashdash#10]
I'd like to be able to set a default value within `dashdash.addOptionType(...)` so that it doesn't need to be set on each individual option that uses that type.

**Comment by trentm** [comment]
Sounds reasonable. Do you have a concrete/realistic example where you'd use that? I find that helps a lot for motivation in examples, tests, etc. The best contrived example I have is a 'fruit' type where a user wants to default it a '-p FRUIT, --pie FRUIT' option to 'apple'. :)

**Comment by marshall007** [comment]
Yea sure, my use-case is that I have a couple different arguments that take a semver-style version number and I'd like them all to default to `v0.0.0`. Here's the code:

```js
dashdash.addOptionType({
  name: 'semver',
  takesArg: true,
  // default: '0.0.0',
  helpArg: 'v0.0.0',
  parseArg: function(opt, optstr, val) {
    var [ ..., major, minor, patch ] = val.match(/v?(\d+)\.(\d+)\.(\d+)/);
    return "v" + major + "." + minor + "." + patch;
  }
});
```

**Comment by trentm** [comment]
published dashdash@1.9.0
Thanks.
